File: Diagrammatize.Design.Pattern/code/cxx/Composite/A2/File.cpp
#include "File.hpp"
#include <iostream>

File::File(const std::string& name_, int size_): name(name_), size(size_) {
}

std::string File::getName() const {
    return name;
}

int File::getSize() const {
    return size;
}

void File::printList(const std::string& prefix) const {
    std::cout << prefix << "/" << *this << std::endl;
}

File: Diagrammatize.Design.Pattern/code/python/Composite/A2/File.py
from Entry import Entry

class File(Entry):
    def __init__(self, name, size):
        super().__init__()
        self.name = name
        self.size = size

    def getName(self):
        return self.name

    def getSize(self):
        return self.size

    def printListAux(self, prefix):
        print(prefix + "/" + str(self))

